### DISCUSSION TITLE
Custom events with underscores do not work, right?

### DISCUSSION THREAD

**Discussion by pohlt** [alpinejs/alpine#3023] [1. Help] [answered]
When I use `$dispatch` I often split multi-word event names with underscores. For some reason, this doesn't work.

[Here's a test page.](https://codepen.io/pohlt/pen/QWmyYNO)

Is this a limitation of alpine.js or a general JavaScript/web limitation?

Cheers! 🙏🏼

**Comment by acentrist** [comment, marked as answer]
You cannot add event handler in HTML with underscores _ as it will not compile.
However, you can do it through JS of parent component:
If you want to handle an event in HTML of parent, you should directly add prefix in attribute on event.

I would recommend not to create event-types with underscores as it may not even be supported from javascript in future releases

**Reply by pohlt** [reply]
Would it make sense for alpine to issue a warning or just crash if a `$dispatch` gets an event with underscores?

**Reply by SimoTod** [reply]
Isn't it just because the _ is missing from the reg exp? https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives.js#L171

I don't think there's a reason why it shouldn't be allowed. As far as I know, there are no restrictions in the HTML specs about underscores in events and attribute names

**Reply by gbaggaley** [reply]
Is there any way around this at all? I'm using Gravity Forms and all of the events it fires (e.g. https://docs.gravityforms.com/gform_confirmation_loaded/) include underscores

**Reply by ekwoka** [reply]
Opened PR: https://github.com/alpinejs/alpine/pull/3756

**Comment by ekwoka** [comment]
This was just merged and released, in case you didn't see
